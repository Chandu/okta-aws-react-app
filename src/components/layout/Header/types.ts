import { AnyStyledComponent } from 'styled-components';

export interface StaticProps {
    LeftSide: AnyStyledComponent;
    RightSide: AnyStyledComponent;
}
