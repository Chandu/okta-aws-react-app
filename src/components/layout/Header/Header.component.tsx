import { WithStyle } from '@medly-components/utils';
import React from 'react';
import * as Styled from './Header.styled';
import { StaticProps } from './types';

export const Header: React.FC & WithStyle & StaticProps = props => {
    return <Styled.Header {...props} />;
};

Header.displayName = 'Header';
Header.Style = Styled.Header;
Header.LeftSide = Styled.LeftSide;
Header.RightSide = Styled.RightSide;

export default Header;
