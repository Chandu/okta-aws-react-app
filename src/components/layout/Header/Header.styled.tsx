import { styled } from '@utils';

export const Header = styled('header')`
    box-sizing: border-box;
    height: 72px;
    background-color: ${({ theme }) => theme.colors.white};
    border-top-left-radius: 0.8rem;
    padding: 0 4rem 0 2.4rem;
    display: flex;
    align-items: center;
    justify-content: space-between;
    position: relative;
    border-bottom: ${({ theme }) => `1px solid ${theme.colors.grey[200]}`};
`;

export const LeftSide = styled('div')`
    display: flex;
    align-items: center;
    border-bottom: 1px solid transparent;
    & > * {
        margin-right: ${({ theme }) => theme.spacing.S2};
    }
`;

export const RightSide = styled('div')`
    display: flex;
    align-items: center;
    & > * {
        margin-right: 0;
        margin-left: ${({ theme }) => theme.spacing.S2};
    }
`;
