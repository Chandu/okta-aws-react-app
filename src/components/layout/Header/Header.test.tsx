import { render } from '@test-utils';
import React from 'react';
import Header from './Header.component';

describe('Header', () => {
    it('should render properly', () => {
        const { container } = render(
            <Header>
                <Header.LeftSide>Left SIde</Header.LeftSide>
                <Header.RightSide>Right SIde</Header.RightSide>
            </Header>
        );
        expect(container).toMatchSnapshot();
    });
});
