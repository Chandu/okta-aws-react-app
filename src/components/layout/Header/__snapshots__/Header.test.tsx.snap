// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  height: 72px;
  background-color: #ffffff;
  border-top-left-radius: 0.8rem;
  padding: 0 4rem 0 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  border-bottom: 1px solid #dfe4e9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid transparent;
}

.c1 > * {
  margin-right: 0.8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > * {
  margin-right: 0;
  margin-left: 0.8rem;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      Left SIde
    </div>
    <div
      class="c2"
    >
      Right SIde
    </div>
  </header>
</div>
`;
